packages/core/src/server/devServer.ts

```diff
@@ -231,6 +231,7 @@ export async function startDevServer<
         const closeServer = async () => {
           await devMiddlewares.close();
           httpServer.close();
+          await options.context.hooks.onRestartDevServerHook.call();
```

Comment:
this hook should not called in closeServer function, it not consistent with semantics.